sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -522,146 +882,339 @@ protected override async ValueTask InitializeProcessingForPartitionAsync(Partiti
+        /// <param name="partitionId">The identifier of the Event Hub partition the task is associated with.  Events will be read only from this partition.</param>
+        /// <param name="startingPosition">The position within the partition where the task should begin reading events.</param>
+        /// <param name="maximumWaitTime">The maximum amount of time to wait to for an event to be available before emitting an empty item; if <c>null</c>, empty items will not be published.</param>
+        /// <param name="retryOptions">The set of options to use for determining whether a failed operation should be retried and, if so, the amount of time to wait between retry attempts.</param>
+        /// <param name="trackLastEnqueuedEventProperties">Indicates whether or not the task should request information on the last enqueued event on the partition associated with a given event, and track that information as events are received.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
         ///
-        /// <exception cref="EventHubsClientClosedException">Occurs when this <see cref="EventProcessorClient" /> instance is already closed.</exception>
-        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessEventAsyncHandler" /> or <see cref="ProcessErrorAsyncHandler" /> set.</exception>
+        /// <returns>The running task that is currently receiving and processing events in the context of the specified partition.</returns>
         ///
-        public override Task StartAsync()
-        {
-            Argument.AssertNotClosed(IsClosed, nameof(EventProcessorClient));
-
-            lock (StartProcessorGuard)
+        private Task RunPartitionProcessingAsync(string partitionId,
+                                                 EventPosition startingPosition,
+                                                 TimeSpan? maximumWaitTime,
+                                                 EventHubsRetryOptions retryOptions,
+                                                 bool trackLastEnqueuedEventProperties,
+                                                 CancellationToken cancellationToken = default) => Task.Run(async () =>
             {
-                if (_processEventAsyncHandler == null)
+                var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                var taskCancellationToken = cancellationSource.Token;
+
+                ActivePartitionProcessorTokenSources[partitionId] = cancellationSource;
+
+                // Context is set to default if operation fails.  This shouldn't fail unless the processor tries processing
+                // a partition it doesn't own.
+
+                PartitionContexts.TryGetValue(partitionId, out var context);
+
+                var clientOptions = new EventHubConsumerClientOptions
```

Comment:
You're going to use the same sets of options across all partitions;  maybe consider having a class-level member to create them once and share them?

Reply:
I'm not sure how much of the remnant base-concrete class separation we should keep to avoid extra work in the future. This method was intended to be kept in the base class, while the concrete class makes calls to it with user-specified parameters. So we could have different options in the original scenario.

I'll be keeping it like this before moving stuff, but let me know if you prefer me to trim all the base-related decisions we made. They should be internal, so not exactly a critical concern.

Reply:
It's very unlikely that we'll be able to derive this instance from our future base.   I'd look at it as a stand-alone class.  Changing the inheritance chain would be, itself, a breaking change.